localstack/services/stepfunctions/asl/component/state/state_execution/state_map/iteration/distributed_iteration_component.py

```diff
@@ -202,6 +202,8 @@ def _eval_body(self, env: Environment) -> None:
             raise ex
         finally:
             env.event_history = execution_event_history
+            self._eval_input = None
+            self._workers.clear()
```

Comment:
I'm assuming that `self._job_pool.await_jobs()` within `_map_run` ensures that all workers finished execution here, so we can safely clear the `_workers` list.

Reply:
Indeed!